packages/cli/src/lib/snapshot/snapshot.spec.ts

```diff
@@ -0,0 +1,84 @@
+jest.mock('../platform/authenticatedClient');
+jest.mock('fs');
+
+import {createReadStream, ReadStream} from 'fs';
+import {join} from 'path';
+import {Readable} from 'stream';
+import {mocked} from 'ts-jest/utils';
+import {AuthenticatedClient} from '../platform/authenticatedClient';
+import {SnapshotFactory} from './snapshotFactory';
+
+const mockedCreateReadStream = mocked(createReadStream);
+const mockedAuthenticatedClient = mocked(AuthenticatedClient);
+const mockedCreateSnapshotFromFile = jest.fn();
+
+const doMockReadStream = () => {
+  mockedCreateReadStream.mockImplementation(() => {
+    const buffer = Buffer.from('this is a tést');
+    const readable = new Readable();
+    readable._read = () => {};
+    readable.push(buffer);
+    readable.push(null);
+    return readable as unknown as ReadStream;
+  });
+};
+
+const doMockAuthenticatedClient = () => {
+  mockedAuthenticatedClient.mockImplementation(
+    () =>
+      ({
+        getClient: () =>
+          Promise.resolve({
+            resourceSnapshot: {createFromFile: mockedCreateSnapshotFromFile},
+          }),
+      } as unknown as AuthenticatedClient)
```

Comment:
I don't understand why you have to do such cast 🤔

Reply:
So I don't have to implement the missing properties
![image](https://user-images.githubusercontent.com/12199712/120254210-4f1a5080-c257-11eb-91d4-170df75504ed.png)